jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -492,11 +506,16 @@ public void signalEvent(String type, Object event) {
 			        }
 				}
 				if (((org.jbpm.workflow.core.WorkflowProcess) getWorkflowProcess()).isDynamic()) {
-					for (Node node : getWorkflowProcess().getNodes()) {
+				    Node [] nodes = getWorkflowProcess().getNodes();
+				    // OCRAM: UGLY
+				    boolean stackless = isStackless();
```

Comment:
Because in short, execution order in stackless (because it's stackless, and not recursive) is the "reverse" of what happens with the old/recursive execution model. It's the same logic that happens with the node instance trigger (actions). 

BTW, I've since made this logic a lot more pretty.